code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -273,9 +271,10 @@
 /mob/living/simple_animal/gib()
 	if(icon_gib)
 		flick(icon_gib, src)
-	if(meat_amount && meat_type)
-		for(var/i = 0; i < meat_amount; i++)
-			new meat_type(src.loc)
+	if(butcher_results)
+		for(var/path in butcher_results)
+			for(var/i = 1; i <= butcher_results[path];i++)

```

Comment:
@Crazylemon64 @Spacemanspark 
```
for loop proc
See also: 
    break proc 
    continue proc 
    do proc 
    for list proc 
    while proc 
Format: 
    for(Init, Test, Inc) Statement 
```
Styling. Semicolons are only an alternative method, because byond is horrible at deciding what it wants it's syntax to be.